### DISCUSSION TITLE
Space Combat Expansion

### DISCUSSION THREAD

**Discussion by Valo56** [zunath/SWLOR_NWN#1477] [Ideas]
Emphasis here: This isn't about changing any of the basic math or functionality behind starfighter combat. That all works fine as is. We don't need a revamp to the revamp. Also, you already have enough on your plate. I'll contribute to this through coding any way I can, though some things are well beyond my ability to handle as of yet.

Rather, this is to encourage teamwork between players, de-emphasise high alpha-strike damage as the be-all-end-all of space combat, and to make different ships of the same tier equally viable.

The suggestion, 
-The pilot of a ship has one or two high slots only, based on the classification of their ship.
-If their ship has two high slots, they have less turret gunners.
-If they have one high slot, they have more turret gunners.
-Turret gunners get their own weapon slots.
-The pilot's main guns of the ship are high damage, long cooldown weapons like turbolasers, ion cannons and concussion missiles. They also get support items like hull and shield repairers.
-The turret gunners have fast firing, low damage weapons like ion repeaters, point defense weapons, proton torpedoes, and other such weapons.
-Choosing between something like a throne and a consular becomes a choice of how many players you can call on to help crew your ship. If you can fill out all those turret gunners, great, get a Consular with its extra turret slots. If not, stick with the Throne since you'll have two main guns.
-Provide added functionality to players not at the pilot seat and not in a turret. Whenever the ship takes damage to its armor, debris or blue/red fire placeables will appear throughout the ship, interactable. Players can use their engineering skill to repair the ship. Maybe include [Spare Ship Parts] items as a consumable to speed up the process. This way players can play as ship engineers.
-Maybe increase the overall health and shields of all players and NPC's by at least 3, while keeping damages the same. That way space combat is a more prolonged affair, like ground combat, where it's a matter of maintaining resources and cooperating with each other, rather than just blasting through everything in sight.
-Everyone doing actions aboard the ship gets experience in piloting. Engineers repairing the ship from the inside get experience in engineering. 

Side suggestions:
-I want to add PC starfighters soon, within the next month. Is it possible to set up ships to have default, unchangeable modules? For example, when I make these ships, I want bombers to have a huge bonus to explosive damage. Any way I can set them up so they have a custom module by default that can't be removed?
-Ability to make dockhands work aboard a player starship would be great, if they were set up such that only certain ships (the PC starfighters I want to add) can land in them. Would this be possible? Ideally it would work like a normal dockhand and then if you press /exit (or a new command /dock) while within a few meters of something like a Throne or Consular with a dockhand aboard it you'd dock there, and then you could speak with the dockhand to launch and it'd spawn your ship out in space where the mothership is.

What I can do:
I can make new items, I can adjust stats, I can make new NPC's. I can adjust the existing ships' loadouts. That's all easy, leave it to me.

What I can't do:
I have no idea how to go about adding functionality to turret gunners. I have no idea how to cause the placeables to spawn in the ship when the ship takes damage. The whole dockhand and static module loadout is also something I can't do.

**Comment by JohnWildkins** [comment]
I want to look into helping with this (namely turret gunning) once I'm done with ACP.

**Comment by Valo56** [comment]
I was thinking a way of limiting the tier of module that can be equipped to a ship would be good. Primary weapons will be heavy hitters and I don't think a t1 ship should be able to equip a t5 turbolaser. I'm not worried about the tiers of other equipment though
